src/libs/context.js

```diff
@@ -153,6 +160,23 @@ function nameOrFallback () {
   return this.name
 }
 
+/**
+ * determine if an input has a user-defined name
+ */
+function hasGivenName () {
+  if (
+    this.name &&
+    typeof this.name === 'string' &&
+    this.name !== `${this.type}_${this.id}` &&
+    this.name !== `${this.type}_${this.defaultId}` &&
+    // radio and checkbox options have their value as part of their ID so we need to filter those out too
+    this.name !== `${this.type}_${(String(this.id).replace('_' + String(this.value), ''))}`
+  ) {
+    return true
+  }
+  return false
```

Comment:
```suggestion
return typeof this.name !== 'boolean'
```

Reply:
I believe this is effectively the same and prevents us from blocking people calling fields by a name that happens to conflict with our internal naming mechanism. i.e. someone happens to have chosen to call their name `${field_name}_${id}`

if `name === true` then we know the name is default, if `name === false` then we know they dont want us to use a name. if `name === 'foobar'` then we know they want their own name.

Reply:
so I would definitely do that, except that this.name is computed and does get set to a String value in some cases. If there’s a way to capture the original prop value and not depend on the computed value that would work.

Reply:
This.name is a prop, this.context.name is a computed prop. A bit confusing